src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Validation/IAsyncValidatableObject.cs

```diff
@@ -0,0 +1,19 @@
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Validation
+{
+    /// <summary>
+    /// Represents a contract to validate an object in async fashion.
+    /// </summary>
+    public interface IAsyncValidatableObject : IValidatableObject
```

Comment:
I'm not sure about deriving from `IValidatableObject`: Since we don't want to use sync-forced async, the object can't be used where an `IValidatableObject` can. And then `Person` and the base class is not needed.